src/components/@shared/FormInput/InputElement.tsx

```diff
@@ -159,6 +159,15 @@ export default function InputElement({
             disabled={disabled}
             {...field}
             {...props}
+            onKeyPress={
+              type === 'number'
+                ? (event) => {
+                    if (!/[0-9.]/.test(event.key)) {
```

Comment:
We might consider putting the regex in a constant, something like  `numberInputValidation` and use it in both places

Reply:
Also, do we really need to use onKeyPress? wouldn't a `pattern` attribute on input work in this case?

Reply:
also could not believe it but yes, `pattern` does not prevent users from entering letters, it just triggers the `:valid/:invalid` pseudo classes on the input.

In any case, Claudia is right with reducing duplication, but [as said before](https://github.com/oceanprotocol/market/pull/1261#issuecomment-1080499602), if we decide to put this on every input, then it should be set on `DefaultInput`.

Also with "number" in this case we mean all numbers users can enter and this regex is only allowing integers, or decimals in US notation. We want all the notations